tests/CMakeLists.txt

```diff
@@ -11,6 +11,20 @@ if (NOT AUTOPAS_BUILD_TESTS)
     return()
 endif ()
 
+option(AUTOPAS_ENABLE_COVERAGE "Enable code coverage, requires GCC and CMAKE_BUILD_TYPE=Debug." OFF)
+message(STATUS "Enable code coverage: " ${AUTOPAS_ENABLE_COVERAGE})
+
+if (AUTOPAS_ENABLE_COVERAGE)
+  if (NOT AUTOPAS_BUILD_TESTS OR NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+    message(WARNING "Disabling code coverage.\n CMAKE_BUILD_TYPE=Debug and AUTOPAS_BUILD_TESTS=ON required.")
+    set(AUTOPAS_ENABLE_COVERAGE OFF CACHE BOOL "Enable code coverage, requires GCC and CMAKE_BUILD_TYPE=Debug." FORCE)
+    message(STATUS "Enable code coverage: " ${AUTOPAS_ENABLE_COVERAGE})
+  endif()
+  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    message(FATAL_ERROR "AUTOPAS_ENABLE_COVERAGE only works with GCC. Change your compiler or set AUTOPAS_ENABLE_COVERAGE=OFF")
+  endif()
```

Comment:
in this case you should also force `AUTOPAS_ENABLE_COVERAGE=OFF`

Reply:
done in [889f266](https://github.com/AutoPas/AutoPas/pull/888/commits/889f26618f64e9d66805dd320b2ef8d07f14578a)